src/com/amplitude/Event.java

```diff
@@ -3,7 +3,104 @@
 import org.json.simple.JSONObject;
 
 public class Event {
+    // should move into Constants.java
+    public static final int MAX_PROPERTY_KEYS = 1024;
+    public static final int MAX_STRING_LENGTH = 1000;
+    public static final String TAG = "com.amplitude.Event"; //AmplitudeClient.class.getName();
 
-    private JsonObject eventProps;
+    private JSObject event;
+
+    public Event(String eventName, JSONObject eventProps, JSONObject groups, JSONObject groupProps,
+                 JSONObject userProps, String app_version,int event_id, int session_id, String insert_id,
+                 String user_id, String device_id, boolean outOfSession, long timestamp) {
+        this.event = new JSONObject();
+        try {
+            this.event.put("eventName", eventProps.getString("eventName"));
+            this.event.put("eventProps", (eventProps == null) ? new JSONObject() : truncate(eventProps));
+
+            this.event.put("groups", (groups == null) ? new JSONObject() : truncate(groups));
+            this.event.put("groupProps", (groups == null) ? new JSONObject() : truncate(groups));
+
+            this.event.put("outOfSession", (outOfSession == true) ?  true : false);
+            this.event.put("timestamp", timestamp);
+
+            this.event.put("userProps",(userProps == null) ? new JSONObject() : truncate(userProps));
+            this.event.put("user_id", replaceWithJSONNull(user_id));
+            this.event.put("device_id", replaceWithJSONNull(device_id));
+            this.event.put("uuid", UUID.randomUUID().toString());
+            this.event.put("session_id", (this.outOfsession) ? -1 : session_id); // session_id = -1 if outOfSession = true;
```

Comment:
Need some solution to automatically set session id, that's unique to the user. Maybe a text file somewhere storing the session id number? How do the other SDKs do it? UNIX milliseconds time on start of session?